frontend/src/features/admin-form/responses/ResponsesPage/SecretKeyVerification.tsx

```diff
@@ -0,0 +1,196 @@
+import { useCallback, useMemo, useRef } from 'react'
+import { RegisterOptions, useForm } from 'react-hook-form'
+import { BiUpload } from 'react-icons/bi'
+import {
+  Container,
+  FormControl,
+  FormErrorMessage,
+  IconButton,
+  Input,
+  Skeleton,
+  Stack,
+  Text,
+} from '@chakra-ui/react'
+import simplur from 'simplur'
+
+import { useIsMobile } from '~hooks/useIsMobile'
+import formsgSdk from '~utils/formSdk'
+import Button from '~components/Button'
+import FormLabel from '~components/FormControl/FormLabel'
+import Link from '~components/Link'
+
+import { FormActivationSvg } from '~features/admin-form/settings/components/FormActivationSvg'
+
+import { useResponsesContext } from './ResponsesContext'
+
+const SECRET_KEY_NAME = 'secretKey'
+const SECRET_KEY_REGEX = /^[a-zA-Z0-9/+]+={0,2}$/
+
+interface SecretKeyFormInputs {
+  [SECRET_KEY_NAME]: string
+}
+
+const useSecretKeyVerification = () => {
+  const { setSecretKey, formPublicKey, isLoading, responsesCount } =
+    useResponsesContext()
+
+  const {
+    formState: { errors },
+    setError,
+    register,
+    setValue,
+    handleSubmit,
+  } = useForm<SecretKeyFormInputs>()
+
+  const fileUploadRef = useRef<HTMLInputElement | null>(null)
+
+  const secretKeyValidationRules: RegisterOptions = useMemo(() => {
+    return {
+      required: "Please enter the form's secret key",
+      pattern: {
+        value: SECRET_KEY_REGEX,
+        message: 'The secret key provided is invalid',
+      },
+      validate: (secretKey) => {
+        // Should not see this error message.
+        if (!formPublicKey) return 'This form is not a storage mode form'
```

Comment:
I'm seeing this error every time I try to upload a secret key, might need to add a mock public key to `createMockForm`